core/includes/common.inc

```diff
@@ -1281,6 +1290,36 @@ function backdrop_http_request($url, array $options = array()) {
   return $result;
 }
 
+/**
+ * Determine whether to strip sensitive headers from a request when redirected.
+ *
+ * @param string $url
+ *   The url from the original outbound http request.
+ *
+ * @param string $location
+ *   The location to which the request has been redirected.
+ *
+ * @return boolean
+ *   Whether sensitive headers should be stripped from the request before
+ *   following the redirect.
+ */
+function _backdrop_should_strip_sensitive_headers_on_http_redirect($url, $location) {
+  $url_parsed = parse_url($url);
+  $location_parsed = parse_url($location);
+  if (!isset($location_parsed['host'])) {
+    return FALSE;
+  }
+  $strip_on_host_change = settings_get('backdrop_http_request_strip_sensitive_headers_on_host_change', TRUE);
+  $strip_on_https_downgrade = settings_get('drupal_http_request_strip_sensitive_headers_on_https_downgrade', TRUE);
+  if ($strip_on_host_change && strcasecmp($url_parsed['host'], $location_parsed['host']) !== 0) {
+    return TRUE;
+  }
+  if ($strip_on_https_downgrade && $url_parsed['scheme'] !== $location_parsed['scheme'] && 'https' !== $location_parsed['scheme']) {
```

Comment:
Having the value before the variable when checking `!==` seems odd. Is there any specific reason we are doing it that way, instead of:
```suggestion
  if ($strip_on_https_downgrade && $url_parsed['scheme'] !== $location_parsed['scheme'] && $location_parsed['scheme'] !== 'https') {
```

Reply:
@klonos, the "Yoda style" is used to avoid accidentally using assignment `=` instead of equality comparison `==`. I'm fine changing. See [this explanation](https://stackoverflow.com/questions/25518264/why-do-some-people-put-the-value-before-variable-in-if-statement).